pkg/instrumentation/exporter_test.go

```diff
@@ -0,0 +1,209 @@
+					SecretName: "my-certs",
+					CA:         "ca.crt",
+					Cert:       "cert.crt",
+					Key:        "key.key",
+				},
+			},
+			expected: corev1.Pod{
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "otel-auto-secret-my-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "my-certs",
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "otel-auto-secret-my-certs",
+									ReadOnly:  true,
+									MountPath: "/otel-auto-instrumentation-secret-my-certs",
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OTEL_EXPORTER_OTLP_ENDPOINT",
+									Value: "http://collector:4318",
+								},
+								{
+									Name:  "OTEL_EXPORTER_OTLP_CERTIFICATE",
+									Value: "/otel-auto-instrumentation-secret-my-certs/ca.crt",
+								},
+								{
+									Name:  "OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE",
+									Value: "/otel-auto-instrumentation-secret-my-certs/cert.crt",
+								},
+								{
+									Name:  "OTEL_EXPORTER_OTLP_CLIENT_KEY",
+									Value: "/otel-auto-instrumentation-secret-my-certs/key.key",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "crt and key from secret and ca from configmap",
+			exporter: v1alpha1.Exporter{
+				Endpoint: "http://collector:4318",
+				TLS: &v1alpha1.TLS{
+					SecretName:    "my-certs",
+					ConfigMapName: "ca-bundle",
+					CA:            "ca.crt",
+					Cert:          "cert.crt",
+					Key:           "key.key",
+				},
+			},
+			expected: corev1.Pod{
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "otel-auto-secret-my-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "my-certs",
+								},
+							},
+						},
+						{
+							Name: "otel-auto-configmap-ca-bundle",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "ca-bundle",
+									},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "otel-auto-secret-my-certs",
+									ReadOnly:  true,
+									MountPath: "/otel-auto-instrumentation-secret-my-certs",
+								},
+								{
+									Name:      "otel-auto-configmap-ca-bundle",
+									ReadOnly:  true,
+									MountPath: "/otel-auto-instrumentation-configmap-ca-bundle",
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OTEL_EXPORTER_OTLP_ENDPOINT",
+									Value: "http://collector:4318",
+								},
+								{
+									Name:  "OTEL_EXPORTER_OTLP_CERTIFICATE",
+									Value: "/otel-auto-instrumentation-configmap-ca-bundle/ca.crt",
```

Comment:
Can someone not keep the cert in a secret?

Reply:
certs should be in secret this is how cert-manager and OCP works. 

ca-bundle can be in both.